tests/st/calicoctl/test_crud.py

```diff
@@ -35,745 +35,1087 @@ class TestPool(TestBase):
-                                           'ports': ['100:200'],
-                                           'selector': "type=='application'",
-                                           'tag': 'alphatag'},
-                                       'icmp': {'type': 10, 'code': 6},
-                                       'protocol': 'tcp',
-                                       'source': {
-                                           'notNets': ['10.1.0.0/16'],
-                                           'notPorts': [1050],
-                                           'notSelector': "type=='database'",
-                                           'notTag': 'bartag',
-                                           'nets': ['10.0.0.0/16'],
-                                           'ports': [1234, '10:20'],
-                                           'selector': "type=='application'",
-                                           'tag': "production"}}],
-                      }}),
-        ("profile2", {'apiVersion': 'projectcalico.org/v2',
-                      'kind': 'Profile',
-                      'metadata': {
-                          'name': 'profile2',
-                      },
-                      'spec': {
-                          'egress': [{'action': 'allow',
-                                      'destination': {},
-                                      'source': {}}],
-                          'ingress': [{'ipVersion': 6,
-                                       'action': 'deny',
-                                       'destination': {},
-                                       'source': {}}],
-                      }}),
-    ]
+    def test_replace_with_resource_version(self):
+        """
+        Test that resource version operates correctly with update, i.e.
+        calicoctl honors the resource version when it's specified.
+        """
+
+        # Create a new Network Policy and get it to determine the current
+        # resource version.
+        rc = calicoctl("create", data=networkpolicy_name1_rev1)
+        rc.assert_no_error()
+        rc = calicoctl(
+            "get networkpolicy %s -o yaml" % name(networkpolicy_name1_rev1))
+        rc.assert_no_error()
+        rev0 = rc.decoded
 
-    @parameterized.expand(testdata)
-    def test_create_from_file_yaml(self, name, data):
-        self._check_data_save_load(data)
-        res_type = data['kind']
-        logger.debug("Testing %s" % res_type)
-        # Write out the files to load later
-        self.writeyaml('/tmp/%s-1.yaml' % res_type, data)
+        # Replace the Network Policy (with no resource version) and get it to
+        # assert the resource version is not the same.
+        rc = calicoctl("replace", data=networkpolicy_name1_rev2)
+        rc.assert_no_error()
+        rc = calicoctl(
+            "get networkpolicy %s -o yaml" % name(networkpolicy_name1_rev2))
+        rc.assert_no_error()
+        rev1 = rc.decoded
+        self.assertNotEqual(rev0, rev1)
```

Comment:
the same comment applies here (and rest of the tests), `resourceVersion` is not explicitly being asserted (different from what the comment says)

Reply:
ah oops.

I've fixed other instances and squashed.  just waiting for tests to pass